packages/sdk/src/across/clients/bridgePool.ts

```diff
@@ -96,30 +97,77 @@ class PoolState {
   }
 }
 
+type EventIdParams = { blockNumber: number; transactionIndex: number; logIndex: number };
 export class PoolEventState {
+  private seen = new Set<string>();
+  private iface: ethers.utils.Interface;
   constructor(
     private contract: bridgePool.Instance,
     private startBlock = 0,
     private state: bridgePool.EventState = bridgePool.eventStateDefaults()
-  ) {}
+  ) {
+    this.iface = new ethers.utils.Interface(bridgePool.Factory.abi);
+  }
+  private makeId(params: EventIdParams) {
+    return [params.blockNumber, params.transactionIndex, params.logIndex].join("!");
+  }
+  hasEvent(params: EventIdParams) {
+    return this.seen.has(this.makeId(params));
+  }
+  private addEvent(params: EventIdParams) {
+    return this.seen.add(this.makeId(params));
+  }
+  private filterSeen = (params: EventIdParams) => {
+    const seen = this.hasEvent(params);
+    if (!seen) this.addEvent(params);
+    return !seen;
+  };
   public async read(endBlock: number) {
     if (endBlock <= this.startBlock) return this.state;
     const events = (
       await Promise.all([
         ...(await this.contract.queryFilter(this.contract.filters.LiquidityAdded(), this.startBlock, endBlock)),
         ...(await this.contract.queryFilter(this.contract.filters.LiquidityRemoved(), this.startBlock, endBlock)),
       ])
-    ).sort((a, b) => {
-      if (a.blockNumber < b.blockNumber) return -1;
-      if (a.transactionIndex < b.transactionIndex) return -1;
-      if (a.logIndex < b.logIndex) return -1;
-      return 1;
-    });
+    )
+      .filter(this.filterSeen)
+      .sort((a, b) => {
+        if (a.blockNumber < b.blockNumber) return -1;
+        if (a.transactionIndex < b.transactionIndex) return -1;
+        if (a.logIndex < b.logIndex) return -1;
```

Comment:
Does this work for primary + secondary sorting?

For instance, what happens if `a.blocknumber > b.blockNumber`, but `a.transactionIndex < b.transactionIndex`?

I think the correct implementation is something more like:
```ts
if (a.blockNumber !== b.blockNumber) return a.blockNumber - b.blockNumber;
if (a.transactionIndex !== b.transactionIndex) return a.transactionIndex - b.transactionIndex;
...
```

Reply:
ohh wow yea you are right. good catch!